src/client/api/object.c

```diff
@@ -315,22 +298,20 @@ daos_obj_anchor_split(daos_handle_t oh, uint32_t *nr, daos_anchor_t *anchors)
 
 	/** TBD - support more than per shard iteration */
 	if (*nr != 0 && *nr != layout->ol_nr) {
-		D_ERROR("For now, num anchors should be the same as what is"
-			" reported as optimal\n");
+		D_ERROR("For now, num anchors should be the same as what is reported as optimal");
 		D_GOTO(out, rc = -DER_INVAL);
 	}
 
 	*nr = layout->ol_nr;
 
 	if (anchors) {
-		int		grp_size;
-		uint32_t	i;
+		int grp_size;
 
 		rc = dc_obj_get_grp_size(oh, &grp_size);
 		if (rc)
 			D_GOTO(out, rc);
 
-		for (i = 0; i < layout->ol_nr; i++) {
+		for (uint32_t i = 0; i < layout->ol_nr; i++) {
```

Comment:
this is not our coding style

Reply:
We don't have much code that looks like this but keeping the scope of variables as small as possible is something that is explicitly stated in our style guide and it's something that's generally considered best-practice across the industry.  I know I've fixed at least one bug in our codebase where a loop variable was misused later in the function.

Changing existing code is laborious and time consuming, plus there are the issues it causes with glt blame and patches across branches etc but I that doesn't change the fact this style would be safer, more concise and preferable overall.

Reply:
i do not disagree. but we are not going to go on and change every part of the code to do this.
we are just going to cause more disruptions to ongoing prs, feature branches that is not going to give that much gain.

so i will approve the change for this file, because i know there is not much work going on in the object api area. @jxiong but please refrain from pushing such a change more broadly before discussing this more in developer channels and/or gatekeeping channels.

Reply:
@mchaarawi are you in the community slack channel?  My understanding is this PR was created to support a discussion there.

Reply:
I am. this is not a community question though. this is developer specific topic.